client/blocks/signup-form/passwordless.jsx

```diff
@@ -184,11 +186,57 @@ class PasswordlessSignupForm extends Component {
 		goToNextStep();
 	};
 
-	onInputChange = ( { target: { value } } ) =>
+	handleEmailDomainSuggestionError = ( email, emailSuggestion ) => {
+		this.props.recordTracksEvent( 'calypso_signup_domain_suggestion_generated', {
+			original_email: JSON.stringify( email ),
+			email_suggestion: JSON.stringify( emailSuggestion ),
+		} );
+		this.setState( {
+			email,
+			errorMessages: [
+				this.props.translate( 'Did you mean {{emailSuggestion/}}?', {
+					components: {
+						emailSuggestion: (
+							<Button
+								plain={ true }
+								className="signup-form__domain-suggestion-confirmation"
+								onClick={ () => {
+									this.setState( {
```

Comment:
Can we extract it from the inline event into its function? Will help readability and reusability.